types/index.d.ts

```diff
@@ -622,6 +622,27 @@ export function renameKeys<MAP extends Dictionary<string>>(
   obj: OBJ
 ) => PickRenameMulti<MAP, OBJ>;
 
+/**
+ * Creates a new object with the own properties of the provided object, but a
+ * single key is renamed according to the currentKey as `'oldKey'` and newKey as `'newKey'`.
+ * When some key is not found in the currentKey, then it's passed as-is.
+ */
+
+function renameKey<
+  KEYS extends keyof OBJ,
+  OBJ extends { readonly [s in KEYS]: any }
+>(currentKey: KEYS, newKey: string): (obj: OBJ) => OBJ;
+
+function renameKey<
+  KEYS extends keyof OBJ,
+  OBJ extends { readonly [s in KEYS]: any }
+>(currentKey: KEYS): (newKey: string) => (obj: OBJ) => OBJ;
+
+function renameKey<
+  KEYS extends keyof OBJ,
+  OBJ extends { readonly [s in KEYS]: any }
+>(currentKey: KEYS, newKey: string, obj: OBJ): OBJ;
```

Comment:
We need a typing tests as well. Here is how it looks like for `renameKeys`: https://github.com/char0n/ramda-adjunct/blob/5cd86a4f2964800e3b8c3b6947bda444ee991f75/types/test/renameKeys.ts